### ISSUE TITLE
Plugin Install Command

### ISSUE THREAD

**Issue by carolynvs** [deislabs/porter#771]
We have a command to install mixins, `porter mixin install`. Now that we have plugins, we need a similar command that installs plugins:

```
porter plugin install NAME [--version VERSION] [--feed-url FEED_URL] [--url URL]
```

* NAME is the name of the plugin binary, for example `azure`. It is a required argument.

The flags should work just like they do for the mixin install command:
* `--version|v` is the version of the plugin to install. It defaults to `latest` and is optional.
* `--feed-url` is the url of an atom feed that contains the plugin. It defaults to the deislabs plugin feed `https://cdn.deislabs.io/porter/plugins/atom.xml`. This isn't being generated yet but will be soon.
* `--url` is the URL where the binary can be downloaded, using the same name format that the mixins use. It should allow for someone to publish binaries to github releases.

The plugin binary for the current OS/ARCH should be downloaded to PORTER_HOME/plugins/. For example the azure plugin would be downloaded to **~/.porter/plugins/azure**.

The plugin installation should be recorded in **~/.porter/plugins/cache.json** just like we remember which mixins we have installed.

The changes should include:
* The command, see https://github.com/deislabs/porter/blob/3df87f160ba34cd163d3ea5c4fd91d1d8ffdeb87/cmd/porter/mixins.go#L50
* Unit tests

**Comment by salaxander** [comment]
@carolynvs I'm interested in taking this on as a way to get involved and familiarize myself with the project a bit :)

**Comment by carolynvs** [comment]
That would be great! Feel free to ask questions here or in the #porter channel on the CNCF slack if you have any questions. Starting with the plugins is a good area because there are a lot more available issues to grab after this if you have fun on this one. 😀

**Comment by carolynvs** [comment]
Closed by #874
